sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/ResolveDefaultColumnsUtil.scala

```diff
@@ -143,6 +144,7 @@ object ResolveDefaultColumns {
     }
     // Analyze the parse result.
     val plan = try {
+      val analyzer: Analyzer = schema.defaultColumnAnalyzer
```

Comment:
Why do we need to have an analyzer per schema? Can we reply on an object like `SimpleAnalyzer`?

Reply:
We can use an object, I updated the code to do that instead.